src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1108 @@
+
+### Prerequisites
+
+- Familiarity with HTML, CSS, JavaScript.
+- Familiarity with the Adobe Express add-ons environment; if you need a refresher, follow the [quickstart](/guides/getting_started/quickstart.md) guide.
+- Familiarity with the Adobe Express Document API, covered in [this tutorial](grids-addon.md).
+- An Adobe Express account; use your existing Adobe ID or create one for free.
+- Node.js version 16 or newer.
+
+### Topics Covered
+
+<!-- List block here -->
+<ListBlock slots="text1, text2" repeat="2" iconColor="#2ac3a2" icon="disc" variant="fullWidth" />
+
+[Invoking Document Model Sandbox methods from the UI iframe](#proxy-api)
+
+[Invoking UI iframe methods from the Document Model Sandbox](#logic)
+
+[Proxy API](#proxy-api)
+
+[Context Closures](#functions)
+
+![](images/stats-addon-animation.gif)
+
+<InlineAlert slots="text" variant="warning"/>
+
+**IMPORTANT:** The Document Model Sandbox references are currently **experimental only**, so you will need to set `experimentalApis` flag to `true` in the [`requirements`](../../references/manifest/index.md#requirements) section of the `manifest.json` to use them. *Please do not use these APIs in any add-ons you plan to distribute or submit with updates until they have been deemed stable.*  Also, please be aware that you should only test these experimental APIs against non-essential documents, as they could be lost or corrupted.
+
+## Getting Started with the Communication API
+
+As we've seen in the previous Adobe Express [Document API tutorial](grids-addon.md), add-ons belong to the **UI iframe**: a sandboxed environment subject to [CORS policies](../develop/context.md#cors), where the User Interface (UI) and the add-on logic are built. The iframe itself has limited editing capabilities, though: via the `addOnUISdk` module, it can invoke a few methods to import media (image, video, and audio) and export the document into a number of formats, like `.pdf`, `.mp4` or `.jpg` for example.
+
+The **Document API** makes new, more powerful capabilities available, allowing the add-on to manipulate elements directly—like scripting in Desktop applications such as Photoshop or InDesign. This API is one component of the Document Sandbox, a JavaScript execution environment that also includes a restricted set of Web API (mostly debugging aids) as well as the means for the UI iframe and the Document API to exchange messages—the Communication API. This infrastructure is paramount as it bridges the gap between the two environments, allowing them to create a seamless experience.
+
+### Proxies
+
+How does this all work, then? Not via messaging, as somebody coming from CEP might expect, but rather exposing proxies for the *other context* to use. The process originates in the `runtime` objects (the UI iframe's and the Document Sandbox's), retrieved from their SDK instances:
```

Comment:
I guess these are technically messages under the hood, but I see what you're getting at given that they are so abstracted w/ comlink. Worth noting that new devs coming to build on Express may not know how CEP did things here, or what CEP even is.

Reply:
Yup, rephrasing from clarity, thanks!